browser/brave_content_browser_client.cc

```diff
@@ -223,6 +230,35 @@ void BraveContentBrowserClient::AdjustUtilityServiceProcessCommandLine(
 #endif
 }
 
+std::vector<std::unique_ptr<blink::URLLoaderThrottle>>
+BraveContentBrowserClient::CreateURLLoaderThrottles(
+    const network::ResourceRequest& request,
+    content::BrowserContext* browser_context,
+    const base::RepeatingCallback<content::WebContents*()>& wc_getter,
+    content::NavigationUIData* navigation_ui_data,
+    int frame_tree_node_id) {
+  auto result = ChromeContentBrowserClient::CreateURLLoaderThrottles(
+        request, browser_context, wc_getter, navigation_ui_data,
+        frame_tree_node_id);
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+  const auto* cmd_line = base::CommandLine::ForCurrentProcess();
```

Comment:
can we encapsulate this within a speadreader method call? The network service can be a dependency of speedreader so we can pass the request into it

Reply:
ok, sounds good!

Reply:
Second thought - such encapsulation requires us to always have a dependency on a component and build this "no-op" implementation when speedreader is disabled. Looks like an overkill?

Reply:
I don't think it's overkill, I think the buildflags are fundamentally flawed when used this way because there is no practical way to enforce that you have correctly embedded everything within them. It places too much burden on the "user" of the speedreader code imo and it's much simpler to just check for `nullptr` than to try to make sure every piece of code that touches speedreader has appropriate buildflag guards